bindings/pydeck/pydeck/data_utils/type_checking.py

```diff
@@ -18,3 +7,21 @@ def is_pandas_df(obj):
         Returns True if object is a Pandas DataFrame and False otherwise
     """
     return obj.__class__.__module__ == "pandas.core.frame" and obj.to_records and obj.to_dict
+
+
+def has_geo_interface(obj):
+    return hasattr(obj, "__geo_interface__")
+
+
+def records_from_geo_interface(data):
+    """Un-nest data from object implementing __geo_interface__ standard"""
+    flattened_records = []
+    for d in data.__geo_interface__.get("features"):
```

Comment:
This will be empty if the geo interface returns a plain geometry without a feature.

Reply:
This isn't resolved. A `__geo_interface__` doesn't always have a `features` key when it's a plain geometry.

Reply:
```py
from shapely.geometry import shape

geojson_geometry = {"type": "Point", "coordinates": [125.6, 10.1]}
shapely_geometry = shape(geojson_geometry)
shapely_geometry.__geo_interface__
# {'type': 'Point', 'coordinates': (125.6, 10.1)}
```

I'd argue that this is a valid input case; it just doesn't have properties so you wouldn't be able to do data-driven styling.

Reply:
For at least GeoPandas, it looks like `__geo_interface__` always renders with this top-level `features` key. Let's get GeoPandas working and we'll revisit these other use cases.